res/controllers/Traktor-Kontrol-S2-MK2-hid-scripts.js

```diff
@@ -0,0 +1,1336 @@
+  TraktorS2MK2.linkChannelOutput("[EffectRack1_EffectUnit2]", "group_[Channel2]_enable", TraktorS2MK2.outputChannelCallback);
+
+  engine.makeConnection("[EffectRack1_EffectUnit1]", "focused_effect", TraktorS2MK2.onFocusedEffectChange);
+  engine.makeConnection("[EffectRack1_EffectUnit2]", "focused_effect", TraktorS2MK2.onFocusedEffectChange);  
+  TraktorS2MK2.connectEffectButtonLEDs("[EffectRack1_EffectUnit1]");
+  TraktorS2MK2.connectEffectButtonLEDs("[EffectRack1_EffectUnit2]");
+  
+  engine.makeConnection("[Channel1]", "VuMeter", TraktorS2MK2.onVuMeterChanged);
+  engine.makeConnection("[Channel2]", "VuMeter", TraktorS2MK2.onVuMeterChanged);
+
+  engine.makeConnection("[Channel1]", "loop_enabled", TraktorS2MK2.onLoopEnabledChanged);
+  engine.makeConnection("[Channel2]", "loop_enabled", TraktorS2MK2.onLoopEnabledChanged);
+}
+
+TraktorS2MK2.linkDeckOutputs = function(key, callback) {
+  // Linking outputs is a little tricky because the library doesn't quite do what I want.  But this
+  // method works.
+  TraktorS2MK2.controller.linkOutput("[Channel1]", key, "[Channel1]", key, callback);
+  engine.connectControl("[Channel3]", key, callback);
+  TraktorS2MK2.controller.linkOutput("[Channel2]", key, "[Channel2]", key, callback);
+  engine.connectControl("[Channel4]", key, callback);
+}
+
+TraktorS2MK2.linkChannelOutput = function(group, key, callback) {
+  TraktorS2MK2.controller.linkOutput(group, key, group, key, callback);
+}
+
+TraktorS2MK2.lightGroup = function(packet, output_group_name, co_group_name) {
+  var group_ob = packet.groups[output_group_name];
+  for (var field_name in group_ob) {
+    field = group_ob[field_name];
+    if (field.name[0] === "!") {
+      continue;
+    }
+    if (field.mapped_callback) {
+      var value = engine.getValue(co_group_name, field.name);
+      field.mapped_callback(value, co_group_name, field.name);
+    }
+    // No callback, no light!
+  }
+}
+
+TraktorS2MK2.lightDeck = function(group) {
+  // Freeze the lights while we do this update so we don't spam HID.
+  this.freeze_lights = true;
+  for (var packet_name in this.controller.OutputPackets) {
+    packet = this.controller.OutputPackets[packet_name];
+    TraktorS2MK2.lightGroup(packet, group, group);
+    // Shift is a weird key because there's no CO that it is actually associated with.
+    TraktorS2MK2.outputCallback(0, group, "!shift");
+  }
+
+  this.freeze_lights = false;
+  // And now send them all.
+  for (packet_name in this.controller.OutputPackets) {
+    var packet_ob = this.controller.OutputPackets[packet_name];
+    packet_ob.send();
+  }
+}
+
+TraktorS2MK2.init = function(id) {
+  TraktorS2MK2.registerInputPackets();
+  TraktorS2MK2.registerOutputPackets();
+//   var data = [0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x00, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f];
```

Comment:
remove debug or move it to a debugLights function

Reply:
I deleted the debugLights function because that didn't work and it was easier to write this than figure out why debugLights didn't work.

Reply:
ok but what I'm saying is, either delete this or create a new debugLights based on this snippet.  I don't want to have commented-out stuff in the committed file

Reply:
I'd rather have commented out code than a function that confusingly isn't used.

Reply:
how about:

```
DEBUG_LIGHTS = false

debugLights = function() { } 

init = function() {
  ...
  if (DEBUG_LIGHTS) {
    debugLights();
  }
  ...
}
```

Reply:
(obviously pseudojs)

Reply:
That would be more cumbersome to use. As it is, I only have to uncomment these lines and comment the call to `TraktorS2MK2.registerOutputPackets()` right above it.

Reply:
In my fix, you only have to change false to true to enable the debugging, so the usage is not cumbersome.  We need to treat our javascript more like real code, because it is real code. We've made some progress but we need to make sure that our attitude for controller scripts is more like our attitude for C++. That includes fixing linter errors and not leaving debug code sprinkled around, commented. Please fix this as I described.

Reply:
https://github.com/mixxxdj/mixxx/pull/2179#issuecomment-505642498